File: example_bots/java/KnockoutBot.java
import java.util.ArrayList;

/**
 * This bot scans the board, for the lowest cash stack,
 * and will raise the pot to match, in an attempt to
 * knock out the player with the lowest stack.
 */
public class KnockoutBot {
    public static final String BOT_NAME = "Knockout bot";

    public static int act(Observable obs) throws Exception {
        ArrayList<PlayerInfo> players = obs.getActivePlayers();

        int lowestRaise = Integer.MAX_VALUE;

        for (PlayerInfo player : players) {
            if(player.getStack() > 0) {
                lowestRaise = Math.min(lowestRaise, player.getStack() + player.getSpent());
            }
        }
        int maxRaise = obs.getMaxRaise();

        // if the lowest stack is greater than the max raise, then raise the max raise
        if (lowestRaise > maxRaise) {
            return maxRaise;
        } else {
            return lowestRaise;
        }

    }
}

File: example_bots/python/knockout_bot.py
from poker_game_runner.state import Observation
from poker_game_runner.utils import Range, HandType
"""
  This bot scans the board, for the lowest cash stack, and will raise the pot to match, in an attempt to knockout the player.
"""


class Bot:
    def get_name(self):
        return "knockout_bot"

    def act(self, obs: Observation):
        stack_sizes = []
        players = obs.get_active_players()

        for player in players:
            if player.stack == 0:  #removes 0 from stack sizes, to avoid folding
                continue

            stack_sizes.append(player.stack + player.spent)

        lowest_stack = min(stack_sizes)
        max_raise = obs.get_max_raise()
        action = min(lowest_stack, max_raise)  #make sure raise is possible

        return action
